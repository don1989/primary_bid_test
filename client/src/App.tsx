import { useState, useEffect } from 'react';
import './App.css';
import { getUrls } from './requests/urls';
import { AxiosResponse } from 'axios';
import { IGetResponseBody, IUrlPair } from './types';
import ListItem from './components/ListItem';
import SubmissionBox from './components/SubmissionBox';

function App() {
	const [urlList, setUrlList] = useState<IUrlPair[]>([]);

	const refreshList = () => {
		getUrls()
			.then((res: AxiosResponse<IGetResponseBody>) => {
				setUrlList(res.data.urls);
			})
			.catch((err) => {
				console.error(err);
				// TODO: display, an unexpected error occurred
			});
	};

	useEffect(() => {
		refreshList();
	}, []);

	const renderUrlList = () => {
		return (
			<div>
				{urlList.map((urlItem) => (
					<ListItem key={urlItem.longUrl} urlPair={urlItem} />
				))}
			</div>
		);
	};

	return (
		<div className="App">
			<SubmissionBox onSubmitSuccess={refreshList} />
			{renderUrlList()}
		</div>
	);
}

export default App;
