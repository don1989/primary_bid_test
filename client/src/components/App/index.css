.App {
	display: flex;
	text-align: center;
	justify-content: space-evenly;
	padding: 24px;
	width: calc(100% - 48px);
	height: calc(100% - 48px);
}

.List {
	width: 600px;
	height: 100%;
	overflow-y: auto;
}
