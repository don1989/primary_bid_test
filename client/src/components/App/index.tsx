import { useState, useEffect } from 'react';
import styles from './index.module.css';
import { getUrls } from '../../requests/urls';
import { AxiosResponse } from 'axios';
import { IGetResponseBody, IUrlPair } from '../../types';
import ListItem from '../ListItem';
import SubmissionBox from '../SubmissionBox';

function App() {
	const [urlList, setUrlList] = useState<IUrlPair[]>([]);

	const refreshList = () => {
		getUrls()
			.then((res: AxiosResponse<IGetResponseBody>) => {
				setUrlList(res.data.urls);
			})
			.catch((err) => {
				console.error(err);
				// TODO: display, an unexpected error occurred
			});
	};

	useEffect(() => {
		refreshList();
	}, []);

	const renderUrlList = () => {
		return (
			<div className={styles.list}>
				{urlList.map((urlItem) => (
					<ListItem key={urlItem.longUrl} urlPair={urlItem} />
				))}
			</div>
		);
	};

	return (
		<div className={styles.app}>
			<SubmissionBox onSubmitSuccess={refreshList} />
			{renderUrlList()}
		</div>
	);
}

export default App;
