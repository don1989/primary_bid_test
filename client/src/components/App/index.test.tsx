import { render, screen } from '@testing-library/react';
import App from '.';

test('renders App', () => {
	const { asFragment } = render(<App />);
	expect(asFragment()).toMatchSnapshot();
});
