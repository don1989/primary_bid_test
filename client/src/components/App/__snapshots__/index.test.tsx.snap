// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="root"
    >
      <div>
        <h2>
          Enter long URL
        </h2>
        <input
          type="text"
          value=""
        />
        <button
          disabled=""
        >
          Submit
        </button>
      </div>
    </div>
    <div
      class="list"
    />
  </div>
</DocumentFragment>
`;
